### PR TITLE
feat: review the display of thumbnail in embedded html links - MEED-6874 - Meeds-io/MIPs#124

### PR DESCRIPTION
**Pull request by SaraBoutej**
This PR implements a new way to display thumbnails based on their height and width

### CODE REVIEW

**Review by boubaker — COMMENTED** [review]
As discussed to be reviewed to simplify algorithm without canvas

**Comment by boubaker** [comment]
> As discussed to be reviewed to simplify algorithm without canvas

We can use this algorithm while posting the activity to save the illustration background at the same time as the thumbnail image source. This way, we will not determine the background each time we display the image, but we compute it once for all on `write` time.

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 1]
```diff
@@ -473,17 +474,37 @@ export default {
...
+            image: response.type !== 'video' && response.thumbnail_url || '-',
+            html: response.type === 'video' && response.html || '-',
+            title: response.title || '-',
+            description: response.description || '-',
+            previewHeight: response.thumbnail_height || '-',
+            previewWidth: response.thumbnail_width || '-',
+            default_title: this.getContent(this.inputVal, false),
+            comment: this.getContentNoEmbed(this.inputVal)
+          });
+        }
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]:
```suggestion
        const oembedParams = {
              link: oembedUrl || '-',
              image: response.type !== 'video' && response.thumbnail_url || '-',
              html: response.type === 'video' && response.html || '-',
              title: response.title || '-',
              description: response.description || '-',
              previewHeight: response.thumbnail_height || '-',
              previewWidth: response.thumbnail_width || '-',
              default_title: this.getContent(this.inputVal, false),
              comment: this.getContentNoEmbed(this.inputVal),
        };
        if (response.thumbnail_url 
            && response.thumbnail_height 
            && response.thumbnail_width 
            && Number(response.thumbnail_height) >= Number(response.thumbnail_width)) {
          this.getAverageColor(response.thumbnail_url)
            .then(bgColor => {
              oembedParams.bgColor = bgColor;
              this.setOembedParams(oembedParams);
            });
        } else {
            this.setOembedParams(oembedParams);
        }  
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 2]
```diff
@@ -814,6 +835,45 @@ export default {
...
+          }
+          // ~~ used to floor values
+          rgb.r = ~~(rgb.r/count);
+          rgb.g = ~~(rgb.g/count);
+          rgb.b = ~~(rgb.b/count);
+          this.defaultRGB = `rgb(${rgb.r},${rgb.g},${rgb.b})`;
+        })
+        .catch(() => {
+          this.defaultRGB = 'rgb(231, 231, 231)';
+        });
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]:
```suggestion
          return `rgb(${rgb.r},${rgb.g},${rgb.b})`;
        })
        .catch(e => {
          console.debug('Error while computing image background color', e);
          return 'rgb(231, 231, 231)';
        });
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 3]
```diff
@@ -170,7 +170,8 @@ export default {
     baseUrl: eXo.env.server.portalBaseURL,
     containInvalidUsers: false,
     spaceId: null,
-    backUpMessage: null
+    backUpMessage: null,
+    defaultRGB: 'rgb(231, 231, 231)',
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: useless

```suggestion
```

**Review by boubaker — COMMENTED** [review]
Please use `npm run lint` as well

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 3 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 2 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 1 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

**Review by boubaker — APPROVED** [review]

**Pull request merged**